plutus-core/executables/uplc/Main.hs

```diff
@@ -268,64 +293,72 @@ runApplyToData (ApplyOptions inputfiles ifmt outp ofmt mode) =
                        Right (d :: Data) ->
                            pure $ UPLC.Program () ver $ mkConstant () d
 
+---------------- Benchmarking ----------------
+
+runBenchmark :: BenchmarkOptions -> IO ()
+runBenchmark (BenchmarkOptions inp ifmt semvar timeLim) = do
+  prog <- readProgram ifmt inp
+  let criterionConfig = defaultConfig {reportFile = Nothing, timeLimit = timeLim}
+      cekparams = mkMachineParameters semvar PLC.defaultCekCostModel
+      getResult (x,_,_) = either (error . show) (\_ -> ()) x  -- Extract an evaluation result
+      evaluate = getResult . Cek.runCekDeBruijn cekparams Cek.restrictingEnormous Cek.noEmitter
+      -- readProgam throws away De Bruijn indices and returns an AST with Names;
+      -- we have to put them back to get an AST with NamedDeBruijn names.
+      !term = fromRight (error "Unexpected open term in runBenchmark.") .
+                runExcept @FreeVariableError $ UPLC.deBruijnTerm (UPLC._progTerm prog)
+      -- Big names slow things down
+      !anonTerm = UPLC.termMapNames (\(PLC.NamedDeBruijn _ i) -> PLC.NamedDeBruijn "" i) term
+      -- Big annotations slow things down
+      !unitAnnTerm = force (() <$ anonTerm)
```

Comment:
> What if you remove `() <$`? Maybe it's the `force` that matters here rather than the annotations? `Force` sure is important there.

The annotations definitely make a difference. I was seeing a discrepancy of 2-3% between flat files and textual files and after several hours I realised that they had different types of annotation: making  all the annotations `()` removed the discrepancy.